### PR TITLE
add disk circuit breaker, update deleteModel message format

### PR DESCRIPTION
**Pull request by Zhangxunmt**
Signed-off-by: Xun Zhang <xunzh@amazon.com>

### Description
This is a backport of disk circuit breakers and message format change in the earlier PR.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public class DeleteModelTransportAction extends HandledTransportAction<ActionReq
     static final String TIMEOUT_MSG = "Timeout while deleting model of ";
     static final String BULK_FAILURE_MSG = "Bulk failure while deleting model of ";
     static final String SEARCH_FAILURE_MSG = "Search failure while deleting model of ";
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java:44]: [nit] these should start with lower case.

**Review by dhrubo-os — APPROVED** [review]

File: common/src/main/java/org/opensearch/ml/common/breaker/MLCircuitBreakerService.java [hunk 2]
```diff
@@ -62,6 +62,8 @@ public MLCircuitBreakerService init() {
         // Register memory circuit breaker
         registerBreaker(BreakerName.MEMORY, new MemoryCircuitBreaker(this.jvmService));
         log.info("Registered ML memory breaker.");
+        registerBreaker(BreakerName.DISK, new DiskCircuitBreaker());
```
Comment (ylwu-amzn) [inline common/src/main/java/org/opensearch/ml/common/breaker/MLCircuitBreakerService.java:65]: Suggest set the `diskDir` of `DiskCircuitBreaker` as OpenSearch data folder path. Check https://github.com/ylwu-amzn/ml-commons/blob/custom_model/common/src/main/java/org/opensearch/ml/common/breaker/MLCircuitBreakerService.java#L67, and https://github.com/ylwu-amzn/ml-commons/blob/custom_model/plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java#L225

Comment (Zhangxunmt) [reply common/src/main/java/org/opensearch/ml/common/breaker/MLCircuitBreakerService.java]: cool. Just updated it with the OS data folder.

File: plugin/src/main/plugin-metadata/plugin-security.policy [hunk 3]
```diff
@@ -20,4 +20,8 @@ grant {
     permission java.util.PropertyPermission "java.library.path", "read,write";
     permission java.util.PropertyPermission "*", "read,write";
     permission java.lang.RuntimePermission "shutdownHooks";
+
+    // Circuit Breaker
+    permission java.lang.RuntimePermission "getFileSystemAttributes";
+    permission java.io.FilePermission "/", "read";
```
Comment (ylwu-amzn) [inline plugin/src/main/plugin-metadata/plugin-security.policy:26]: As we only check opensearch data folder, remove this permission?

Comment (Zhangxunmt) [reply plugin/src/main/plugin-metadata/plugin-security.policy]: Good catch. Just removed this line and all tests pass. Pushed the change to the second commit.

**Review by ylwu-amzn — APPROVED** [review]

**Review by dhrubo-os — APPROVED** [review]

**Pull request merged**